lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -795,7 +795,129 @@ func (d *zfs) CreateVolumeFromMigration(vol Volume, conn io.ReadWriteCloser, vol
 
 // RefreshVolume updates an existing volume to match the state of another.
 func (d *zfs) RefreshVolume(vol Volume, srcVol Volume, srcSnapshots []Volume, op *operations.Operation) error {
-	return genericVFSCopyVolume(d, nil, vol, srcVol, srcSnapshots, true, op)
+	// Get target snapshots
+	targetSnapshots, err := vol.Snapshots(op)
+	if err != nil {
+		return fmt.Errorf("Failed to get target snapshots: %w", err)
+	}
+
+	srcSnapshotsAll, err := srcVol.Snapshots(op)
+	if err != nil {
+		return err
+	}
```

Comment:
Minor nit, but I'd reverse the snapshot enumerations to avoid doing a network operation if the source doesn't have any snapshot.